### ISSUE TITLE
Debugger entered--Lisp error: (void-function first)

### ISSUE THREAD

**Issue by tsdh** [juergenhoetzel/babel#11]
I've just wanted to try out the package, and `M-x babel RET test RET` immediately gives this error:

```
Debugger entered--Lisp error: (void-function first)
  first(("English"))
  babel("house")
  (progn (babel "house"))
  eval((progn (babel "house")) t)
  elisp--eval-last-sexp(nil)
  eval-last-sexp(nil)
  funcall-interactively(eval-last-sexp nil)
  call-interactively(eval-last-sexp nil nil)
  command-execute(eval-last-sexp)
```

Byte-compiling shows many other undefined functions:

```
In babel-version:
babel.el:777:12: Warning: ‘interactive-p’ is an obsolete function (as of
    23.2); use ‘called-interactively-p’ instead.

In babel-google-wash:
babel.el:882:8: Warning: ‘beginning-of-buffer’ is for interactive use only;
    use ‘(goto-char (point-min))’ instead.

In end of data:
babel.el:875:4: Warning: the function ‘reduce’ is not known to be defined.
babel.el:808:6: Warning: the function ‘find’ is not known to be defined.
babel.el:760:8: Warning: the function ‘incf’ is not known to be defined.
babel.el:456:28: Warning: the function ‘remove*’ is not known to be defined.
babel.el:444:29: Warning: the function ‘first’ is not known to be defined.
babel.el:408:12: Warning: the function ‘mm-decode-coding-region’ is not known
    to be defined.
```

All but `mm-decode-coding-region` are probably `cl` functions which would mean your package wants a header
``` elisp
;; Package-Requires: ((cl-lib "0.7"))
```
and then a `(require 'cl-lib)` and then you'd use `cl-reduce`, `cl-find`, `cl-first`, etc.  (You should not `(require 'cl)` at runtime because that will pollute the obarray with non-package prefixed functions.)

The XEmacs compatibility function `mm-decode-coding-region` has been removed from Emacs in 2016.  Emacs uses `decode-coding-region`.  Is it possible that you've switched the `;; xemacs compatibility` `if` arms at the beginning of `babel.el`?

**Comment by juergenhoetzel** [comment]
Thanks for the report: These issues should be fixed by #12 and #13 

Unfortunately, the two free backends of `babel` ( "FreeTranslation" "Apertium")  are both offline.

The only way to use babel at the moment is by using a Google-Translation API Key.

**Comment by tsdh** [comment]
Thanks for the quick fix, and too bad that the only free backends are offline.

How about dict.cc and dict.leo.org?

**Comment by tsdh** [comment]
There's still an `incf` call which should be `cl-incf` in `babel-chuncify` which raises an error.

Also, the remaining byte-compiler warnings are probably worth fixing (and easy to do so).

```Compiling file /home/horn/.emacs.d/elpa/babel-20210611.1352/babel.el at Fri Jun 11 22:41:49 2021
Entering directory ‘/home/horn/.emacs.d/elpa/babel-20210611.1352/’

In babel-version:
babel.el:777:12: Warning: ‘interactive-p’ is an obsolete function (as of
    23.2); use ‘called-interactively-p’ instead.

In babel-google-wash:
babel.el:824:8: Warning: ‘beginning-of-buffer’ is for interactive use only;
    use ‘(goto-char (point-min))’ instead.

In babel-libretranslate-wash:
babel.el:879:8: Warning: ‘beginning-of-buffer’ is for interactive use only;
    use ‘(goto-char (point-min))’ instead.

In end of data:
babel.el:760:8: Warning: the function ‘incf’ is not known to be defined.
```

**Comment by juergenhoetzel** [comment]
> There's still an `incf` call which should be `cl-incf` in `babel-chuncify` which raises an error.
> 
> Also, the remaining byte-compiler warnings are probably worth fixing (and easy to do so).

Thanks again: These issues are fixed by #15 

`babel.el` is to a large extent  still in a old state (imported from a CVS Repo 12 years ago) and  can certainly use some refactoring :eyes:
